core/src/test/java/org/apache/james/core/MailAddressTest.java

```diff
@@ -47,6 +48,7 @@
                 GOOD_ADDRESS,
                 GOOD_QUOTED_LOCAL_PART,
                 "server-dev@james-apache.org",
+                "a..b@domain.com",
```

Comment:
hmm I don't know according to wikipedia  that's not a valid address because the dot occurs twice in a row which seems to match [RFC5322](https://datatracker.ietf.org/doc/html/rfc5322#section-3.2.3) this would need to be quoted to be a valid local part since it is not a dot-atom

For reference examples from wikipedia : 
good addresses: 
- Abc@example.com
- Abc@10.42.0.1
- Abc.123@example.com
- user+mailbox/department=shipping@example.com
- !#$%&'*+-/=?^_`.{|}~@example.com
- "Abc@def"@example.com
- "Fred Bloggs"@example.com
- "Joe.\\Blow"@example.com
- Loïc.Accentué@voilà.fr8
- pelé@exemple.com
- δοκιμή@παράδειγμα.δοκιμή
- 我買@屋企.香港
- 二ノ宮@黒川.日本
- медведь@с-балалайкой.рф
- संपर्क@डाटामेल.भारत


bad addresses
- Abc.example.com
- Abc.@example.com
- Abc..123@example.com

Reply:
Yes I know. Sorry it is indeed misleading.

> (but maybe that's on purpose to demonstrate the issue ?)

Yes it is, as my fix suggests to move it to bad address/